core/src/main/java/org/kiwix/kiwixmobile/core/main/NightModeViewPainter.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2020 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.main
+
+import android.graphics.ColorMatrixColorFilter
+import android.graphics.Paint
+import android.view.View
+import org.kiwix.kiwixmobile.core.NightModeConfig
+import javax.inject.Inject
+
+/**
+ * NightModeViewPainter class is used to apply respective filters to the views
+ * depending whether the app is in dark mode or not
+ * Created by yashk2000 on 24/03/2020.
+ */
+
+class NightModeViewPainter @Inject constructor(
+  private val nightModeConfig: NightModeConfig
+) {
+
+  private val invertedPaint =
+    Paint().apply { colorFilter = ColorMatrixColorFilter(KiwixWebView.NIGHT_MODE_COLORS) }
+
+  fun <T : View> update(
+    view: T,
+    shouldActivateCriteria: ((T) -> Boolean) = { true },
+    vararg additionalViews: View
+  ) {
+    if (nightModeConfig.isNightModeActive()) {
+      if (shouldActivateCriteria(view)) {
+        for (view in additionalViews) {
+          activateNightMode(view)
```

Comment:
you aren't using additionalViews as additionalViews, you are using it as "the only views that get painted"
I would rewrite de/activate to also accept varargs, and then you can invoke them like so activateNightMode(view, *additionalViews) by using the spread operator

Reply:
@macgills if we use activateNightMode(view, *additionViews), then do we need to use varargs?
In this case, we have both the views we need already, view will have the webview and we can replace the varargs directly with a single view which will contain either the videoView of the holder.content.

Reply:
The design of the api is such that it can handle an arbitrary number of views when calling update.

additionalViews should only be used in the case of the web&videoView and I wouldn't be shocked if more views got added given how this feature grows

Reply:
Ok. Thanks for the explanation on both the comments. 
And sorry for the delay in this pr. I am a bit unfamiliar with the current method we're using. So I'm trying to properly learn what is happening. Thanks a lot for the explanations and the blog posts you linked.

Reply:
They are not easy topics, just keep in mind that what we are trying to do is keep the usage of `painter` as painless as possible. We only specify something if we absolutely must specify it

Reply:
@macgills updated, please review.